lib/barcelona/plugins/secure_instance_plugin.rb

```diff
@@ -0,0 +1,81 @@
+# SecureInstance plugin
+# This plugin configure container instance and bastion instance more secure
+# Usage: bcn district put-plugin discrict1 secure_instance
+
+module Barcelona
+  module Plugins
+   
+    class SecureInstancePlugin < Base
+
+      def on_container_instance_user_data(_instance, user_data)
+        user_data.extend SecureUserData
+        user_data
+      end
+
+      def on_network_stack_template(_stack, template)
+        bastion_lc = template["BastionLaunchConfiguration"]
+        return template if bastion_lc.nil?
+
+        user_data = InstanceUserData.load_or_initialize(bastion_lc["Properties"]["UserData"])
+        user_data.extend SecureUserData
+        bastion_lc["Properties"]["UserData"] = user_data.build
+        template
+      end
+
+      module SecureUserData
+        def self.extended(obj)
+          obj.configure_security
+        end
+
+        def configure_security
+          apply_security_update_on_the_first_boot
+          install_tools_for_pcidss
+        end
+
+        def apply_security_update_on_the_first_boot
```

Comment:
So container instances reboot on the first launch? I don't think we need this for container instances because Amazon releases security-patched AMI periodically

Reply:
basically I don't want to update existing packages for container instances because package updates might break compatibility and make instances stop functioning. For safety, We've chosen to just use plain ECS AMI and not to update yum packages.

Reply:
Actually we had `yum update --security` in container instances but removed that because there was a problem that caused ECS disconnection (If I remember correctly, dockerd died)

Reply:
I see, I will use AMI as is for container instances.

But I will update bastion instances because, while testing it, I have seen the newest AMI has high level vulnerabilities for a week until applied the security update.

I think we need a way to apply patches at least for instances in public network in case there are vulnerabilities in network stack of kernel.

Reply:
yes I agree